nodes/dictionary/dictionary_in.py

```diff
@@ -0,0 +1,124 @@
+    key_4: bpy.props.StringProperty(name="", default='Key 5', update=update_node)
+    key_5: bpy.props.StringProperty(name="", default='Key 6', update=update_node)
+    key_6: bpy.props.StringProperty(name="", default='Key 7', update=update_node)
+    key_7: bpy.props.StringProperty(name="", default='Key 8', update=update_node)
+    key_8: bpy.props.StringProperty(name="", default='Key 9', update=update_node)
+    key_9: bpy.props.StringProperty(name="", default='Key 10', update=update_node)
+
+    def sv_init(self, context):
+        self.inputs.new('SvSeparatorSocket', 'Empty')
+        self.outputs.new('SvDictionarySocket', 'Dict')
+        self['update_event'] = False
+
+    def update(self):
+        print(f'Update {self.name}')
+        # Remove unused sockets
+        for sock in self.inputs:
+            if sock.bl_idname != 'SvSeparatorSocket' and not sock.links:
+                self.inputs.remove(sock)
+
+        # add property to new socket and add extra empty socket
+        free_keys = self.keys - set(sock.prop_name for sock in self.inputs if sock.bl_idname != 'SvSeparatorSocket')
+        if list(self.inputs)[-1].links and len(self.inputs) < 11:
+            socket_from = self.inputs[-1].other
+            self.inputs.remove(list(self.inputs)[-1])
+            re_socket = self.inputs.new(socket_from.bl_idname, 'Dict in sock')
+            re_socket.prop_name = free_keys.pop()
+            re_socket.custom_draw = 'draw_socket'
+            re_link = self.id_data.links.new(socket_from, re_socket)
+            re_link.is_valid = True
+            self.inputs.new('SvSeparatorSocket', 'Empty')
+            self['update_event'] = True
+            setattr(self, re_socket.prop_name, re_socket.other.name)
+            self['update_event'] = False
+
+    def draw_socket(self, socket, context, layout):
+        layout.prop(self, 'up', text='', icon='TRIA_UP')
+        layout.prop(self, 'down', text='', icon='TRIA_DOWN')
+        layout.prop(self, socket.prop_name)
+
+    def process(self):
+
+        if not any((sock.links for sock in self.inputs)):
+            return
+
+        max_len = max([len(sock.sv_get()) for sock in list(self.inputs)[:-1]])
+        data = [chain(sock.sv_get(), cycle([None])) for sock in list(self.inputs)[:-1]]
+        keys = [sock.prop_name for sock in list(self.inputs)[:-1]]
+        out = []
+        for i, *props in zip(range(max_len), *data):
+            out_dict = SvDict({getattr(self, key): prop for key, prop in zip(keys, props) if prop is not None})
+            for sock in list(self.inputs)[:-1]:
+                out_dict.inputs[sock.identifier] = {'type': sock.bl_idname,
+                                                    'name': getattr(self, sock.prop_name),
+                                                    'nest': sock.sv_get()[0].inputs
+                                                    if sock.bl_idname == 'SvDictionarySocket' else None}
```

Comment:
I think indentation allows you to write this way, and keep the line short
```python
                out_dict.inputs[sock.identifier] = {
                    'type': sock.bl_idname,
                    'name': getattr(self, sock.prop_name),
                    'nest': sock.sv_get()[0].inputs if sock.bl_idname == 'SvDictionarySocket' else None}
```
it's slightly easier to read